From ff171eff8ea2e65e3f400cf5cb6ce025402381ce Mon Sep 17 00:00:00 2001
From: jackqdyulei <jackqdyulei@google.com>
Date: Tue, 19 Jun 2018 16:52:07 -0700
Subject: [PATCH 35/71] Update PBAP info for all pairing dialogs. DO NOT MERGE

This is a bug from ag/4210612, in which it only update PBAP info
for USER_ENTRY_DIALOG. So in other kind of dialogs it never upload
correct PBAP info to bluetooth backend.

This CL fix it by updating PBAP for all dialogs.

Change-Id: Ia39eee1acaece555e8e5a305ec2c803294d7efbd
Merged-In: Ia39eee1acaece555e8e5a305ec2c803294d7efbd
Bug: 109842273
Bug: 72872376
Test: RunSettingsRoboTests
(cherry picked from commit 7015e20a554fd7f8f7ded7359a5ecc13a82a0464)
---
 .../bluetooth/BluetoothPairingController.java | 22 ++++++++++++-------
 1 file changed, 14 insertions(+), 8 deletions(-)

diff --git a/src/com/android/settings/bluetooth/BluetoothPairingController.java b/src/com/android/settings/bluetooth/BluetoothPairingController.java
index d06cb7e037..96837e45f8 100644
--- a/src/com/android/settings/bluetooth/BluetoothPairingController.java
+++ b/src/com/android/settings/bluetooth/BluetoothPairingController.java
@@ -23,12 +23,16 @@ import android.text.Editable;
 import android.util.Log;
 import android.widget.CompoundButton;
 import android.widget.CompoundButton.OnCheckedChangeListener;
+
 import com.android.settings.R;
 import com.android.settings.bluetooth.BluetoothPairingDialogFragment.BluetoothPairingDialogListener;
 import com.android.settingslib.bluetooth.LocalBluetoothManager;
 import com.android.settingslib.bluetooth.LocalBluetoothProfile;
+
 import java.util.Locale;
 
+import android.support.annotation.VisibleForTesting;
+
 /**
  * A controller used by {@link BluetoothPairingDialog} to manage connection state while we try to
  * pair with a bluetooth device. It includes methods that allow the
@@ -50,8 +54,10 @@ public class BluetoothPairingController implements OnCheckedChangeListener,
 
     // Bluetooth dependencies for the connection we are trying to establish
     private LocalBluetoothManager mBluetoothManager;
-    private BluetoothDevice mDevice;
-    private int mType;
+    @VisibleForTesting
+    BluetoothDevice mDevice;
+    @VisibleForTesting
+    int mType;
     private String mUserInput;
     private String mPasskeyFormatted;
     private int mPasskey;
@@ -82,7 +88,6 @@ public class BluetoothPairingController implements OnCheckedChangeListener,
         mDeviceName = mBluetoothManager.getCachedDeviceManager().getName(mDevice);
         mPbapClientProfile = mBluetoothManager.getProfileManager().getPbapClientProfile();
         mPasskeyFormatted = formatKey(mPasskey);
-
     }
 
     @Override
@@ -96,12 +101,13 @@ public class BluetoothPairingController implements OnCheckedChangeListener,
 
     @Override
     public void onDialogPositiveClick(BluetoothPairingDialogFragment dialog) {
+        if (mPbapAllowed) {
+            mDevice.setPhonebookAccessPermission(BluetoothDevice.ACCESS_ALLOWED);
+        } else {
+            mDevice.setPhonebookAccessPermission(BluetoothDevice.ACCESS_REJECTED);
+        }
+
         if (getDialogType() == USER_ENTRY_DIALOG) {
-            if (mPbapAllowed) {
-                mDevice.setPhonebookAccessPermission(BluetoothDevice.ACCESS_ALLOWED);
-            } else {
-                mDevice.setPhonebookAccessPermission(BluetoothDevice.ACCESS_REJECTED);
-            }
             onPair(mUserInput);
         } else {
             onPair(null);
-- 
2.17.1

